experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wrappers
+
+// This file contains a wrapper for Rekor Log Entries.
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"github.com/go-openapi/runtime"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"io/ioutil"
+)
+
+// This wrapper is meant to be modeled after the comments here
+// https://github.com/project-oak/oak/blob/main/oak_functions/client/rust/src/rekor.rs
+// describing that verifying the log entry entails:
+//  -- verifying the signature in `signedEntryTimestamp`, using Rekor's public key,
+// -- verifying the signature in `body.RekordObj.signature`, using Oak's public key,
+// -- verifying that the content of the body matches the input `endorsement_bytes`.
+// And also possibly validating the inclusion proof
+//
+// TODO look into validating the inclusion proof with
+// logEntryAnon.Verification.ValidateInclusionProof.
+// I'm not sure what to pass for "formats" yet
+// https://github.com/sigstore/rekor/blob/e0b79164f2279ad7c3e723a3ee16afbfcf271188/pkg/generated/models/log_entry.go#L365
+
+func getLogEntryAnonFromFile(rekorLogFilePath string) (*models.LogEntryAnon, error) {
+	// get LogEntry, which is a map from strings to LogEntryAnons
+	logEntryBytes, err := ioutil.ReadFile(rekorLogFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read the rekor log file: %v", err)
+	}
+
+	var logEntry models.LogEntry
+
+	err = json.Unmarshal(logEntryBytes, &logEntry)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal the logEntry from file: %s, err: %v", rekorLogFilePath, err)
+	}
+
+	if len(logEntry) != 1 {
+		return nil, fmt.Errorf("For transparent release, Rekor log entries should contain exactly one UUID: %v", logEntry)
```

Comment:
```suggestion
		return nil, fmt.Errorf("for transparent release, Rekor log entries must contain exactly one UUID: %v", logEntry)
```

see https://datatracker.ietf.org/doc/html/rfc2119

Reply:
fixed